pkg/shared/service.go

```diff
@@ -18,6 +22,103 @@ func NewServiceHandler() *ServiceHandler {
 
 func (sh *ServiceHandler) Handle(ctx context.Context, event sdk.Event) error {
 	fmt.Println("handling object ", event.Object.GetObjectKind().GroupVersionKind().String())
+
+	sharedService := event.Object.(*v1alpha1.SharedService)
+	sharedServiceCopy := sharedService.DeepCopy()
+
+	if sharedService.Spec.ServiceType != strings.ToLower(v1alpha1.KeycloakKind) {
+		return nil
+	}
+
+	if event.Deleted {
+		return nil
+	}
+
+	if sharedService.GetDeletionTimestamp() != nil {
+		return sh.finalizeSharedService(sharedServiceCopy)
+	}
+
+	switch sharedService.Status.Phase {
+	case v1alpha1.SSPhaseNone:
+		sh.initSharedService(sharedServiceCopy)
+	case v1alpha1.SSPhaseAccepted:
+		sh.createKeycloaks(sharedServiceCopy)
+	case v1alpha1.SSPhaseComplete:
+		//ToDo What to do here?
+	}
+
+	return nil
+}
+
+func (sh *ServiceHandler) createKeycloaks (sharedService *v1alpha1.SharedService) error {
+	for i := 1; i <= sharedService.Spec.MinInstances; i++ {
```

Comment:
This should  compare against required_instances, rather than min_instances.

I think it might be better to do something like:
```
if numKeycloaks() < sharedService.Spec.RequiredInstance {
   sdk.Create(newKeycloak(sharedService))
}
```

So that one new keycloak is created per loop, and it doesn't get bogged down in API calls if someone sets requiredInstances to a stupid number. Also this logic will continue to work right into the future, if someone goes and changes the value of requiredInstances after it already exists.

Reply:
What's the purpose of min Instances?. I was assuming min instances meant "I want this number running regardless of how many keycloaks i really need". What's updating this required instances value?

Yes, we can have a single keycloak created in a single iteration though, makes sense.

Reply:
MinInstances is the lowest threshold, even if requiredInstances is set to a lower number.

i.e. requiredInstances might be 0, but we'd never go below 1.